core/src/main/java/com/linecorp/armeria/client/AbstractRuleBuilder.java

```diff
@@ -197,6 +201,17 @@ public AbstractRuleBuilder onUnprocessed() {
         return onException(UnprocessedRequestException.class);
     }
 
+    /**
+     * Adds the specified {@code responseDurationNanosFilter}.
+     */
+    public AbstractRuleBuilder onResponseDurationNanos(
```

Comment:
How about adding `onResponseDuration(BiPredicate<? super ClientRequestContext, ? super Duration>)` as well? We could delegate to `onResponseDurationNanos()`.

Reply:
> Can we generalize it a little bit more?

Looks good to me. Let me update this PR! Thank you for your comment!